### PR TITLE
adding test commit changes

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by amnn — APPROVED** [review]

File: sdk/typescript/src/builder/TransactionBlock.ts [hunk 1]
```diff
@@ -624,16 +624,22 @@ export class TransactionBlock {
...
         const dryRunResult = await expectProvider(
           provider,
         ).dryRunTransactionBlock({
           transactionBlock: this.#blockData.build({
-            overrides: { gasConfig: { budget: String(MAX_GAS), payment: [] } },
+            overrides: {
+              gasConfig: {
+                budget: String(MAX_GAS),
+                // use user provided payment if it exists.
+                payment: this.#blockData.gasConfig.payment
```
Comment (Jordan-Mysten) [inline sdk/typescript/src/builder/TransactionBlock.ts:636]: FWIW this was intentionally empty to guarantee this succeeds regardless of the payment that's attached to it or not. I don't have a strong preference for if this should be the case or not, but that was the original idea behind it.

File: sdk/typescript/src/builder/TransactionBlock.ts [hunk 2]
```diff
@@ -624,16 +624,22 @@ export class TransactionBlock {
...
+                  ? this.#blockData.gasConfig.payment
+                  : [],
+              },
```
Comment (666lcz) [inline sdk/typescript/src/builder/TransactionBlock.ts:639]: Can you revert this change? Actually, I don't think we should override the payment object because we use `MAX_GAS` on line 634.

File: sdk/typescript/test/e2e/object-vector.test.ts [hunk 3]
```diff
@@ -27,6 +27,7 @@ describe('Test Move call with a vector of objects as input', () => {
       transactionBlock: tx,
       options: {
         showEffects: true,
+        showBalanceChanges: true,
```
Comment (666lcz) [inline sdk/typescript/test/e2e/object-vector.test.ts:30]: nit: we can remove this

**Review by 666lcz — APPROVED** [review]
